enhancements/microshift/microshift-router-configuration-performance.md

```diff
@@ -0,0 +1,212 @@
+last-updated: 2024-10-07
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-4091
+---
+
+# MicroShift router Operations & performance configuration options
+
+## Summary
+MicroShift's default router is created as part of the platform, but does not
+allow configuring any of its specific parameters. For example, you cannot
+specify the policy for HTTP traffic compression or enable HTTP2 protocol.
+
+In order to allow these operations and many more, a set of configuration options
+is proposed.
+
+## Motivation
+
+Microshift Customers need a way to override the default Ingress Controller  configuration parameters similar as OpenShift does.
+
+
+### User Stories
+
+#### User Story 1
+
+My application starts processing requests from clients, but the connection is
+getting closed before it can respond.
+
+I set `ingress.tuningOptions.serverTimeout` in the configuration file to a
+higher value to accommodate the slow response from the server.
+
+#### User Story 2
+
+The router has many connections open because an application running on my
+cluster doesn't close connections properly.
+
+I set `ingress.tuningOptions.serverTimeout` and
+`spec.tuningOptions.serverFinTimeout` in the ingresscontroller API to a lower
+value, forcing those connections to close sooner if my application stops
+responding to them.
+
+### Goals
+Allow users to configure the additional HAProxy/Router performance customization parameters, see Proposal table for details.
+
+
+### Non-Goals
+N/A
+
+## Proposal
+
+Microshift don't use ingress [operator](https://github.com/openshift/cluster-ingress-operator) , all the customization performed through  configuration file.
+the configuration will propagate to the router deployment [manifest](https://github.com/openshift/microshift/blob/aea40ae1ee66dc697996c309268be1939b018f56/assets/components/openshift-router/deployment.yaml) Environment variables.
+
+see the table below for the proposed configuration changes:
+
+| new configuration                                 | description                                                                                                                                                                                     | default |
+| ------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------- |
+| ingress.httpCompressionMimeTypes                  | list of MIME types that should have compression applied. <br>At least one MIME type must be specified.<br>                                                                                      | none    |
+| ingress.forwardedHeaderPolicy                     | specify when and how the Ingress Controller sets the `Forwarded`, `X-Forwarded-For`, `X-Forwarded-Host`, `X-Forwarded-Port`, `X-Forwarded-Proto`, and `X-Forwarded-Proto-Version` HTTP headers. | append  |
```

Comment:
Not sure if it matters, but "Append" is capitalized (like "Log", and "Respond"), in the PR for this EP